src/transaction_builder.js

```diff
@@ -462,15 +467,17 @@ TransactionBuilder.prototype.__build = function (allowIncomplete) {
   this.inputs.forEach(function (input, i) {
     var scriptType = input.redeemScriptType || input.prevOutType
     if (!scriptType && !allowIncomplete) throw new Error('Transaction is not complete')
-
     var result = buildInput(input, allowIncomplete)
 
     // skip if no result
-    if (!result && allowIncomplete) return
-    if (result && result.script) return tx.setInputScript(i, result.script)
-    if (result && result.witness) return tx.setWitness(i, result.witness)
+    if (!allowIncomplete) {
+      if (SIGNABLE.indexOf(result.type) === -1 && result.type !== bscript.types.P2WPKH) {
+        throw new Error(result.type + ' not supported')
+      }
```

Comment:
Normally we'd end up with [p2pkh, p2pk, multisig], but p2wpkh is just p2pkh, so when saving our knowledge about the inner-most script, we set the P2PKH solution (yep, solveOutput) for scriptCode if it's P2WPKH.